src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -78,51 +84,61 @@ public void initialize(SearchRequestBuilder request) {
 
     @Override
     public SqlElasticSearchRequestBuilder explain() throws SqlParseException {
-        Hint scrollHint = null;
-        for (Hint hint : select.getHints()) {
-            if (hint.getType() == HintType.USE_SCROLL) {
-                scrollHint = hint;
-                break;
-            }
-        }
-        if (scrollHint != null && scrollHint.getParams()[0] instanceof String) {
-            return new SqlElasticSearchRequestBuilder(new SearchScrollRequestBuilder(client,
-                    SearchScrollAction.INSTANCE, (String) scrollHint.getParams()[0])
-                    .setScroll(new TimeValue((Integer) scrollHint.getParams()[1])));
-        }
+        Objects.requireNonNull(this.sqlRequest, "SqlRequest is required for ES request build");
+        buildRequest();
+        checkAndSetScroll();
+        return new SqlElasticSearchRequestBuilder(request);
+    }
 
+    private void buildRequest() throws SqlParseException {
         this.request = new SearchRequestBuilder(client, SearchAction.INSTANCE);
         setIndicesAndTypes();
-
         setFields(select.getFields());
         setWhere(select.getWhere());
         setSorts(select.getOrderBys());
-        setLimit(select.getOffset(), select.getRowCount());
-
-        if (scrollHint != null) {
-            if (!select.isOrderdSelect()) {
-                request.addSort(FieldSortBuilder.DOC_FIELD_NAME, SortOrder.ASC);
-            }
-            request.setSize((Integer) scrollHint.getParams()[0])
-                    .setScroll(new TimeValue((Integer) scrollHint.getParams()[1]));
-        } else {
-            request.setSearchType(SearchType.DFS_QUERY_THEN_FETCH);
-        }
         updateRequestWithIndexAndRoutingOptions(select, request);
         updateRequestWithHighlight(select, request);
         updateRequestWithCollapse(select, request);
         updateRequestWithPostFilter(select, request);
         updateRequestWithInnerHits(select, request);
+    }
 
-        return new SqlElasticSearchRequestBuilder(request);
+    private void checkAndSetScroll() {
```

Comment:
Do we have UT to test the logic in here. IMO, the  setting, fetchSize, rowCount has different combination which determine the scroll logic which should be covered in the UT.

Reply:
Will add unit test for this.

Reply:
Added.